src/main/java/SpringJPA/PageController.java

```diff
@@ -115,6 +115,15 @@ public String makeAPICall(Principal principal, @ModelAttribute String placeholde
         return "redirect:/user";
     }
 
+    @PostMapping("/user/endSub")
+    public String endSub(Principal principal, @ModelAttribute String placeholder){
+        User user = userRepository.findByUsername(principal.getName());
+        user.setRole(UserType.ROLE_NONTRIAL);
+        user.setApiCallCount(0);
```

Comment:
should this not be setApiCallLimit()? we want to set their limit to 0, not their current call count amount

Reply:
Their limit doesn't matter no? Why would the limit need to be set if they can't use the API anyways? Either way, users have default 1000 limit when they're created no matter the role. So I don't think this is necessary.

Reply:
well why would you reset their current call count to 0 at all. You should never reset that value manually, because that value just means how many times they have called it in the last 24h period. The only time that value should be reset is by the timed event which happens at midnight to reset everyone's call counts from the last 24h

Reply:
Understandable, I'll remove that